src/core/achievements.cpp

```diff
@@ -3741,12 +3740,21 @@ static TinyString GetLoginEncryptionMachineKey()
 
   ret.resize(machine_guid_length);
   RegCloseKey(hKey);
-#elif !defined(__ANDROID__)
-#ifdef __linux__
+#else
+#if defined(__linux__)
   // use /etc/machine-id on Linux
   std::optional<std::string> machine_id = FileSystem::ReadFileToString("/etc/machine-id");
   if (machine_id.has_value())
     ret = std::string_view(machine_id.value());
+#elif defined(__APPLE__)
+  // use gethostuuid(2) on macOS
+  const struct timespec ts = { .tv_sec = 2, .tv_nsec = 0 };
```

Comment:
Is 2 seconds wait time reasonable? If this is actually an expensive operation, I'd be concerned about "random" slowdowns to due to machine load causing a different key to be used and invalidating a user's login.

Reply:
I'd just make it infinite.